UI/script.js

```diff
@@ -0,0 +1,354 @@
+    }
+}
+
+class View{
+    _ads;
+    _tmpAd;
+    _username;
+    _isVendor;
+    constructor(adList, username, isVendor){
+        this._ads = new AdsList(adList);
+        this._tmpAd = document.querySelector(".offer-tmp");
+        this._username = username;
+        this._isVendor = isVendor;
+    }
+    showUser(){
+        if(this._username == null || this._username.length == 0){
+            document.querySelector(".profile").textContent = "Log in";
+            document.querySelector(".log-out").style.visibility = "hidden";
+        }
+        else {
+            document.querySelector(".profile").textContent = this._username;
+            document.querySelector(".log-out").style.visibility = "visible";
+        }
+    }
+    _getOffer(ad){
+        let template = document.importNode(this._tmpAd, true);
+        template.querySelector(".image-in-offer").setAttribute('src', ad.photoLink);
+        template.querySelector(".percent").textContent = ad.discount + "%";
+        template.querySelector(".date-expiration").textContent = ad.validUntil.getFullYear() + " " + (ad.validUntil.getMonth() + 1) + " " + ad.validUntil.getDate();
+        template.querySelector(".offer-label").textContent = ad.label;
+        template.querySelector(".seller-name").textContent = ad.vendor;
+        template.querySelector(".seller-link").setAttribute("href", ad.link);
+        if(this._username == null || this._username.length == 0){
+            template.querySelectorAll(".button-offer").forEach(button => button.style.visibility = "hidden");
+            template.querySelector(".add-comment").style.visibility = "hidden";
+        }
+        else if (this._isVendor == true) {
+            template.querySelector(".add-comment").style.visibility = "hidden";
+        }
+        else {
+            template.querySelectorAll(".button-offer").forEach(button => button.style.visibility = "hidden");
+        }
+        template.querySelector(".description-offer").textContent = ad.description;
+        template.querySelector(".rating-number").textContent = ad.rating;
+        let hashtagsTmp = template.querySelector(".hashtags-offer");
+        template.querySelectorAll(".tag").forEach(tag => tag.remove());
+        ad.hashTags.forEach(tag => {
+            let hashtag = document.createElement('span');
+            hashtag.className = "tag";
+            hashtag.textContent = "#" + tag;
+            hashtagsTmp.append(hashtag);
+        });
+        let reviewsTmp = template.querySelector(".reviews-offer");
+        template.querySelectorAll(".review").forEach(rev => rev.remove());
+        ad.reviews.forEach(review => {
+            let review1 = document.createElement('span');
+            review1.className = "review";
+            review1.textContent = review;
+            reviewsTmp.append(review1);
+        });
+        return template;
+    }
+    showAds(){
+        let secondButton = document.querySelector(".offers-list .button-load-more:last-child");
```

Comment:
Let's not use last-child and similar selectors: you may change your markup one day (eg add one more button for some reason) and your JS code will be broken.